From 1a06a98c4372d3f22bc0e1e7bce9aa8464660c61 Mon Sep 17 00:00:00 2001
From: James Lee <james@james.tl>
Date: Tue, 18 Apr 2023 20:10:20 -0400
Subject: [PATCH 1/3] Enable font hinting with UI scaling

Font hinting provides a noticeable resolution bump even on HiDPI
displays.
---
 ui/gfx/font_render_params_linux.cc | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/ui/gfx/font_render_params_linux.cc b/ui/gfx/font_render_params_linux.cc
index 6198dd9c85c9b..579a7de515ac0 100644
--- a/ui/gfx/font_render_params_linux.cc
+++ b/ui/gfx/font_render_params_linux.cc
@@ -224,21 +224,6 @@ FontRenderParams GetFontRenderParams(const FontRenderParamsQuery& query,
     params.hinting = FontRenderParams::HINTING_FULL;
     params.subpixel_rendering = FontRenderParams::SUBPIXEL_RENDERING_NONE;
     params.subpixel_positioning = false;
-  } else if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
-                 switches::kDisableFontSubpixelPositioning)) {
-#if BUILDFLAG(IS_CHROMEOS)
-    // We want to enable subpixel positioning for fractional dsf.
-    params.subpixel_positioning =
-        std::abs(std::round(actual_query.device_scale_factor) -
-                 actual_query.device_scale_factor) >
-        std::numeric_limits<float>::epsilon();
-#else
-    params.subpixel_positioning = actual_query.device_scale_factor > 1.0f;
-#endif  // BUILDFLAG(IS_CHROMEOS)
-
-    // To enable subpixel positioning, we need to disable hinting.
-    if (params.subpixel_positioning)
-      params.hinting = FontRenderParams::HINTING_NONE;
   }
 
   // Use the first family from the list if Fontconfig didn't suggest a family.
-- 
2.37.4

